app/main.py

```diff
@@ -1,3 +1,53 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> float:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance) -> Distance:
+        if not isinstance(other, Distance):
+            distance = Distance(self.km + other)
+        else:
+            distance = Distance(self.km + other.km)
+        return distance
+
+    def __iadd__(self, other: Distance) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, num: int) -> Distance:
+        distance = Distance(self.km * num)
+        return distance
+
+    def __truediv__(self, num: int) -> Distance:
+        distance = Distance(round(self.km / num, 2))
+        return distance
+
+    def __lt__(self, num: int) -> bool:
```

Comment:
provide logick for num type float and Distance

Reply:
sorry, I don't understand what I need to do here – should I rewrite the code so that num type is float/Distance? Or should I just explain why I used "int" as the type? Could you please explain a little more – I'm stuck